### ISSUE TITLE
Usability Testing Prep for Medallia -- Accessibility Review

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#15214]
This issue will be an epic to contain accessibility issues found during the user testing preparation as part of #14945

**Comment by joanneesteban** [comment]
@1Copenut - to verify some of the observed accessibility issues so far, would this sum it up?

- Cognitive challenge for screen reader users. 
  - _Issue:_ Users listen to the entire set of form options using the Forms menu, then make a choice. They're currently hearing the radio/check for every single answer option. (Reads the label and text above it every time, instead of just the label)
- Focus management challenge for screen reader users. 
  - _Issue:_ When the modal opens up, the focus should be set immediately, but it does not. Screen reader users would not know that the modal pops up, and would have to tab through the entire page before being able to close it.
  - _Issue:_ No focus halo on the focus elements. Unclear to users what element has focus.

**Comment by 1Copenut** [comment]
@joanneesteban Hello! I'll reframe your question a bit, and give you the high level items I've noted. These are just first impression notes based on the [VSP defect severity rubric](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md). I'll get turned into proper issue tickets asap. Let me know if you want to discuss further or have questions I can answer.

- **[Defect-severity-1](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)**
  - Focus must be trapped inside the modal dialog. Right now focus remains on the page "behind" the modal and this will cause problems for screen readers.
- **[Defect-severity-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)**
  - Radio groups on the feedback page do not have a focus state. Keyboard users will have a hard time determining where focus is when they're navigating through the form.
  - Focus halo is not shown on the Medallia dialog. This should use the same yellow  focus halo we use everywhere else on VA.gov.
  - VA logo must have alt text
- **[Defect-severity-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)**
  - Powered by text in bottom right has color contrast that's too low. Needs to be at least 4.5:1. Current text is 3.86:1.
  - The Medallia `<iframe>` has an aria-role "dialog" on the `<body>` tag. This is not a correct markup. We would need another div inside the body tag that has this role.
  - Zooming and scaling is disabled on the modal. This prevents mobile device users from pinching or expanding to make the modal bigger or smaller.
  - Radio button labels repeat the legend text every time. This might be too much information.
  - Checkboxes repeat the legend text every time. This makes the checkbox labels confusing.
  - The VA image appears low-resolution (pixelated) on high-density screens, and could reduce Veteran trust in the survey. Can we make this a higher-resolution version or an SVG that is resolution-independent?
- **[Defect-severity-4](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)**
  -  The modal container is not inside an HTML5 landmark region. Could we add a role="region" to contain it? Maybe add an aria-label that could say "Medallia survey modal" or something that effect.
  - Buttons are 34px tall. Would be better if they were closer to 44px tall per Apple HIG or 48px per Google Material guidelines.

**Comment by joanneesteban** [comment]
Thank you!

**Comment by jwoodman5** [comment]
@1Copenut We had a quick chat about this, but I am clarifying in the ticket for reference. There are 2 different surveys referenced in the associated ticket (#14945), which is linked in the description of _this_ ticket. Were both of them part of the 508/accessibility review you conducted here? Thanks.

**Comment by 1Copenut** [comment]
@jwoodman5 I only took a look at the Intercept 9(?) question form, the longer of the two. I have not had a chance to review the Feedback button 3 question.

**Comment by jwoodman5** [comment]
Closing this epic as the contained stories have been closed with fixes and/or issues/trackers for upcoming *should* do enhancements.
